datamover/pkg/drivers/https/prepare.go

```diff
@@ -200,92 +195,17 @@ func getOsdsS3Objs(ctx context.Context, in *pb.RunJobRequest, marker string, lim
 	return rsp.Objects, nil
 }
 
-func getIBMCosObjs(ctx context.Context, conn *pb.Connector, filt *pb.Filter,
-	defaultSrcLoca *LocationInfo) ([]*osdss3.Object, error) {
-	//TODO(acorbellini): reuse getAWSS3Objs function
-	srcObjs := []*osdss3.Object{}
-	objs, err := ibmcosmover.ListObjs(defaultSrcLoca, filt)
-	if err != nil {
-		return nil, err
-	}
-	for i := 0; i < len(objs); i++ {
-		obj := osdss3.Object{Size: *objs[i].Size, ObjectKey: *objs[i].Key, Location: ""}
-		srcObjs = append(srcObjs, &obj)
-	}
-	return srcObjs, nil
-}
-
-func getAwsS3Objs(ctx context.Context, conn *pb.Connector, filt *pb.Filter,
-	defaultSrcLoca *LocationInfo) ([]*osdss3.Object, error) {
-	//TODO:need to support filter
-	srcObjs := []*osdss3.Object{}
-	objs, err := s3mover.ListObjs(defaultSrcLoca, filt)
-	if err != nil {
-		return nil, err
-	}
-	for i := 0; i < len(objs); i++ {
-		obj := osdss3.Object{Size: *objs[i].Size, ObjectKey: *objs[i].Key, Location: ""}
-		srcObjs = append(srcObjs, &obj)
-	}
-	return srcObjs, nil
-}
-
-func getHwObjs(ctx context.Context, conn *pb.Connector, filt *pb.Filter,
-	defaultSrcLoca *LocationInfo) ([]*osdss3.Object, error) {
-	//TODO:need to support filter
-	srcObjs := []*osdss3.Object{}
-	objs, err := obsmover.ListObjs(defaultSrcLoca, filt)
-	if err != nil {
-		return nil, err
-	}
-	for i := 0; i < len(objs); i++ {
-		obj := osdss3.Object{Size: objs[i].Size, ObjectKey: objs[i].Key, Location: ""}
-		srcObjs = append(srcObjs, &obj)
-	}
-	return srcObjs, nil
-}
-
-func getAzureBlobs(ctx context.Context, conn *pb.Connector, filt *pb.Filter,
-	defaultSrcLoca *LocationInfo) ([]*osdss3.Object, error) {
-	srcObjs := []*osdss3.Object{}
-	objs, err := blobmover.ListObjs(defaultSrcLoca, filt)
-	if err != nil {
-		return nil, err
-	}
-	for i := 0; i < len(objs); i++ {
-		obj := osdss3.Object{Size: *objs[i].Properties.ContentLength, ObjectKey: objs[i].Name,
-			Location: ""}
-		srcObjs = append(srcObjs, &obj)
-	}
-	return srcObjs, nil
-}
-
-//to get object details from ceph backend
-func getCephS3Objs(ctx context.Context, conn *pb.Connector, filt *pb.Filter,
-	defaultSrcLoca *LocationInfo) ([]*osdss3.Object, error) {
-	srcObjs := []*osdss3.Object{}
-	objs, err := cephs3mover.ListObjs(defaultSrcLoca, filt)
-	if err != nil {
-		return nil, err
-	}
-	for i := 0; i < len(objs); i++ {
-		obj := osdss3.Object{Size: objs[i].Size, ObjectKey: objs[i].Key, Location: ""}
-		srcObjs = append(srcObjs, &obj)
+func GetMultipartSize() int64 {
+	var size int64 = 16 * 1024 * 1024 // this is the default
+	userSetSize, err := strconv.ParseInt(os.Getenv("PARTSIZE"), 10, 64)
```

Comment:
Was thinking, should the PARTSIZE be coming from ENV or can it be an API to set the PARTSIZE?

Reply:
In my opinion, the size of part should be got from configuration file or from db? if it is a configurable parameter, should it be put into the configuration file?

Reply:
Right now, it's from ENV, see os.Getenv("PARTSIZE"), it's setted in docker-compose.yml.

Reply:
Ok. My opinion was to make the PARTSIZE setting to be configurable from API so that any application using multi-cloud can get the leverage rather than to change YAML or ENV. But that will be an ehancement

Reply:
OK, let me create an enhance issue to trace it.

Reply:
Here is the issue: [#771 ](https://github.com/opensds/multi-cloud/issues/771)